src/Products/Product.php

```diff
@@ -87,10 +88,11 @@ private function set_brand() {
 	 */
 	private function set_category() {
 
-		$return        = '';
-		$taxonomy_name = ShoppingFeedHelper::wc_category_taxonomy();
+		$return           = '';
+		$taxonomy_name    = ShoppingFeedHelper::wc_category_taxonomy();
+		$sf_yoast_options = get_option( Admin\Options::SF_YOAST_OPTIONS );
 
-		if ( class_exists( '\WPSEO_Primary_Term' ) ) {
+		if ( class_exists( \WPSEO_Primary_Term::class ) && ! empty( $sf_yoast_options['use_principal_categories'] ) ) {
```

Comment:
We should use a stricter condition check here : 
`(int) $sf_yoast_options['use_principal_categories'] === 1`

Reply:
i made the change